homeassistant/components/androidtv/config_flow.py

```diff
@@ -0,0 +1,304 @@
+"""Config flow to configure the Android TV integration."""
+import logging
+import os
+import socket
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_HOST, CONF_PORT
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import async_connect_androidtv, validate_state_det_rules
+from .const import (
+    CONF_ADB_SERVER_IP,
+    CONF_ADB_SERVER_PORT,
+    CONF_ADBKEY,
+    CONF_APPS,
+    CONF_EXCLUDE_UNNAMED_APPS,
+    CONF_GET_SOURCES,
+    CONF_SCREENCAP,
+    CONF_STATE_DETECTION_RULES,
+    CONF_TURN_OFF_COMMAND,
+    CONF_TURN_ON_COMMAND,
+    DEFAULT_ADB_SERVER_PORT,
+    DEFAULT_DEVICE_CLASS,
+    DEFAULT_EXCLUDE_UNNAMED_APPS,
+    DEFAULT_GET_SOURCES,
+    DEFAULT_PORT,
+    DEFAULT_SCREENCAP,
+    DEVICE_CLASSES,
+    DOMAIN,
+)
+
+APPS_NEW_ID = "NewApp"
+CONF_APP_DELETE = "app_delete"
+CONF_APP_ID = "app_id"
+CONF_APP_NAME = "app_name"
+
+RESULT_CONN_ERROR = "cannot_connect"
+RESULT_UNKNOWN = "unknown"
+RESULT_SUCCESS = "success"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_file(value):
+    """Validate that the value is an existing file."""
+    file_in = os.path.expanduser(str(value))
+
+    return os.path.isfile(file_in) and os.access(file_in, os.R_OK)
+
+
+def _get_ip(host):
+    """Get the ip address from the host name."""
+    try:
+        return socket.gethostbyname(host)
+    except socket.gaierror:
+        return None
+
+
+class AndroidTVFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize Android TV config flow."""
+        self._host = None
+        self._unique_id = None
+
+    @callback
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(
+                        CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS
+                    ): vol.In(DEVICE_CLASSES),
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                    vol.Optional(CONF_ADBKEY): str,
+                    vol.Optional(CONF_ADB_SERVER_IP): str,
+                    vol.Required(
+                        CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT
+                    ): cv.port,
+                }
+            ),
+            errors=errors or {},
+        )
```

Comment:
```suggestion
    def _show_setup_form(self, user_input=None, error=None):
        """Show the setup form to the user."""
        user_input = user_input or {}
        data_schema = {
            vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
            vol.Required(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS): vol.In(
                DEVICE_CLASSES
            ),
            vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
        }

        if self.show_advanced_options:
            data_schema = {
                vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
                vol.Required(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS): vol.In(
                    DEVICE_CLASSES
                ),
                vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
                vol.Optional(CONF_ADBKEY): str,
                vol.Optional(CONF_ADB_SERVER_IP): str,
                vol.Required(
                    CONF_ADB_SERVER_PORT, default=DEFAULT_ADB_SERVER_PORT
                ): cv.port,
            }

        return self.async_show_form(
            step_id="user",
            data_schema=vol.Schema(data_schema),
            errors={"base": error},
        )
```

Reply:
This could probably be simplified a bit. Since the adb server addon is deprecated, only advanced users should see this. Perhaps some warning should be displayed to the user to suggest moving away from adb server.